src/com/lftechnology/java/training/alina/jdbc/service/UserService.java

```diff
@@ -0,0 +1,324 @@
+            return;
+        case 'c':
+            LOGGER.log(Level.INFO, "\n========================\nEdit own Information\n========================\n");
+            getEditInfo(scanner, employee);
+        case 'd':
+            LOGGER.log(Level.INFO, "\n========================\nUser successfully logged out.\n========================\n");
+            UserDaoImpl userDao = new UserDaoImpl();
+            userDao.checkEmployeeLogin(scanner);
+        default:
+            LOGGER.log(Level.INFO, "\n========================\nInvalid entry, Please choose from menu option.\n========================\n");
+            return;
+        }
+    }
+
+    /**
+     * Gets employee edit informations
+     * 
+     * @param scanner
+     *            {@link Scanner}
+     * @param result
+     *            {@link ResultSet}
+     * @throws SQLException
+     * @author Alina Shakya <alinashakya@lftechnology.com>
+     */
+    private void getEditInfo(Scanner scanner, Employee employee) throws SQLException {
+        char choice = ' ';
+        while (choice != Constants.EMPLOYEE_EDIT_EXIT) {
+            EmployeeEditView.displayEmployeeEditMenu();
+            choice = UtilityService.getSelectedMenu(scanner, "Select an option (a-e) : ");
+            getEmployeeEditOptions(scanner, choice, employee);
+        }
+    }
+
+    /**
+     * Gets employee edit options
+     * 
+     * @param scanner
+     *            {@link Scanner}
+     * @param choice
+     *            {@link Character}
+     * @param result
+     *            {@link ResultSet}
+     * @throws SQLException
+     * @author Alina Shakya <alinashakya@lftechnology.com>
+     */
+    private void getEmployeeEditOptions(Scanner scanner, char choice, Employee employee) throws SQLException {
+        switch (choice) {
+        case 'a':
+            LOGGER.log(Level.INFO, "\n========================\nEdit fullname\n========================\n");
+            employee.setFullname(UtilityService.getInputData(scanner, "Enter new Fullname to Edit : "));
+            EmployeeController.updateEmployeeInfo(Constants.FULLNAME, employee.getFullname(), employee.getEmployeeId());
+            return;
+        case 'b':
+            LOGGER.log(Level.INFO, "\n========================\nEdit Department\n========================\n");
+            employee.setDepartment(UtilityService.getInputData(scanner, "Enter new Department to Edit : "));
+            EmployeeController.updateEmployeeInfo(Constants.DEPARTMENT, employee.getDepartment(), employee.getEmployeeId());
+            return;
+        case 'c':
+            LOGGER.log(Level.INFO, "\n========================\nEdit Address\n========================\n");
+            employee.setAddress(UtilityService.getInputData(scanner, "Enter new Address to Edit : "));
+            EmployeeController.updateEmployeeInfo(Constants.ADDRESS, employee.getAddress(), employee.getEmployeeId());
+            return;
+        case 'd':
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 7 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1151)